src/arrayMethodJoin.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const joinedArr = [];
+    let j = 0;
+    let newSeparator = separator;
+
+    if (newSeparator === undefined) {
+      newSeparator = ',';
+    }
+
+    for (let i = 0; i < (this.length - 1) * 2 - 1; i++) {
+      if (this[j] === null || this[j] === undefined) {
```

Comment:
you have not defined `j` but you are using it here 
and you just need to implement normal join method, so require only normal for loop , you do not require to have loop till  `(this.length - 1) * 2 - 1`